### PR TITLE
Improve README.md

### LINKED ISSUE #62: VM do not get IP address after reboot
After rebooting macOS 14.7.x, you don't get MAC address.

What happened:
- Had working lima vms, stopped before the upgrade
- Upgrade macOS to 14.7.1
- Start the limv vms
  VM do not get IP address

```
$ ip a
...
3: lima0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:55:55:66:93:49 brd ff:ff:ff:ff:ff:ff
    inet6 fd9b:5a14:ba57:e3d3:5055:55ff:fe66:9349/64 scope global dynamic mngtmpaddr noprefixroute 
       valid_lft 2591975sec preferred_lft 604775sec
    inet6 fe80::5055:55ff:fe66:9349/64 scope link 
       valid_lft forever preferred_lft forever
```

Looking in system settings > network > options shows that bootpd is allowed to receive incoming connections.

Checking with socketfilterfw confirm that bootpd is configured:

```
# /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd 
The application is already a part of the firewall 
```

Note: this is managed system. During startup "Cisco Secure Client" application is launched and there is no way to avoid this. I always quit the client but it or some other software that I don't control may break the system configuration in some way.

## Workaround

Fixed by removing bootpd from the firewall and adding it again:

```
# /usr/libexec/ApplicationFirewall/socketfilterfw --remove /usr/libexec/bootpd
Application at path ( /usr/libexec/bootpd ) removed from firewall 

# /usr/libexec/ApplicationFirewall/socketfilterfw --add /usr/libexec/bootpd   
Application at path ( /usr/libexec/bootpd ) added to firewall 

# /usr/libexec/ApplicationFirewall/socketfilterfw --unblock /usr/libexec/bootpd
Incoming connection to the application is permitted 
```

Maybe we should include the --remove part in the README?

### LINKED ISSUE #70: Update docs and launchd configuration - stdout not used
Since #58 we log only to stderr, so /var/log/socket_vmnet/stdout is never used.

- [ ] Remove from README.md
- [ ] Remove from launchd configuration

### PR DESCRIPTION
**Pull request by nirs**
- Simplify make install.* examples
- Update Troubleshooting section (fixes #70)
- Improve recovery instruction when ip address not assigned (fixes #62)
- Improve markdown formatting

[Preview changes](https://github.com/lima-vm/socket_vmnet/blob/a7c199caed10372b6920523787ff14c14a4f10bc/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +46,7 @@ brew install socket_vmnet
 ```
 
 The binaries will be installed onto the following paths:
+
```
Comment (AkihiroSuda) [inline README.md:49]: Is this spacing required by some linter? If so, it should be enforced in the CI

Comment (AkihiroSuda) [reply README.md:49]: BTW what is the benefit of this spacing style?

File: README.md [hunk 2]
```diff
@@ -151,19 +156,21 @@ sudo port unload socket_vmnet
...
-to set the prefix to a directory that can be only written by the root.
-
-Note that `/usr/local/bin` is sometimes chowned for a non-admin user, so `/usr/local` is *not* an appropriate prefix.
+This installs binaries onto the following paths:
 
-The binaries will be installed onto the following paths:
 - `/opt/socket_vmnet/bin/socket_vmnet`
 - `/opt/socket_vmnet/bin/socket_vmnet_client`
 
+You can customize the install location using the `PREFIX` environmemnt variable,
```
[the remark this answers is not in the document]
Comment (nirs) [reply README.md:167]: I can add one example of using PREFIX near this section.

File: README.md (hunk 1 above)
Comment (nirs) [reply README.md:49]: We can use yamllint to enforce this. I'll add an issue to add it.

Comment (nirs) [reply README.md:49]: The source is more clear. The rendered page looks the same.

File: README.md [hunk 3]
```diff
@@ -332,18 +349,26 @@ NOTE: don't confuse MAC addresses of VMs with the MAC address of `socket_vmnet`
...
-- Set environment variable `DEBUG=1`
-- See `${HOMEBREW_PREFIX}/var/log/socket_vmnet/{stdout,stderr}` (when using launchd).
-  The path was previously `${HOMEBREW_PREFIX}/var/run/socket_vmnet.{stdout,stderr}` until March 2023.
+
+- To enable verbose debug logs, set the environment variable `DEBUG=1`.
+- When using launchd, logs are written to `/var/log/socket_vmnet/stderr`.
+  `/var/log/socket_vmnet/stdout` is not used and expected to be empty.
+
+> [!NOTE]
+> Until March 2023, logs were written to `/var/run/socket_vmnet.{stdout,stderr}`.
```
Comment (AkihiroSuda) [inline README.md:374]: Probably this note can be now removed

Comment (nirs) [reply README.md:374]: ok, removing

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**